src/cmd/function.js

```diff
@@ -8,36 +9,32 @@ class FunctionCommand {
 
 	listFunctions() {
 		const api = new ApiClient();
-		if (!api.ready()) {
-			return -1;
-		}
+		api.ensureToken();
 
-		return api.getAllAttributes()
-			.then((devices) => {
+		return api.getAllAttributes().then(devices => {
+			let lines = [];
+			for (let i = 0; i < devices.length; i++) {
 
-				let lines = [];
-				for (let i = 0; i < devices.length; i++) {
+				const device = devices[i];
+				const available = [];
+				if (device.functions) {
 
-					const device = devices[i];
-					const available = [];
-					if (device.functions) {
-
-						for (let idx = 0; idx < device.functions.length; idx++) {
-							const name = device.functions[idx];
-							available.push('  int ' + name + '(String args) ');
-						}
-					}
-
-					let status = device.name + ' (' + device.id + ') has ' + available.length + ' functions ';
-					if (available.length === 0) {
-						status += ' (or is offline) ';
+					for (let idx = 0; idx < device.functions.length; idx++) {
+						const name = device.functions[idx];
+						available.push('  int ' + name + '(String args) ');
```

Comment:
Fancier with `

`  int  ${name}(String args) `

Reply:
Yeah, most of this code predates the backtick operator and I didn't refactor all the instances of string concatenation. It's work left for a future maintainer :)